File: python/src/exampleC.py


class ShoppingBasketBuilder:

    def add_item(self, item):
        raise NotImplementedError()

    def build(self):
        raise NotImplementedError()


class ItemBuilder:
    def __init__(self, name):
        raise NotImplementedError()

    def with_price(self, price):
        raise NotImplementedError()

    def with_quantity(self, quantity):
        raise NotImplementedError()

    def build(self):
        raise NotImplementedError()

File: swift/Sources/swift/exampleC.swift
import Foundation

class ShoppingBasketBuilder {

    func addItem(item: BasketItem) -> ShoppingBasketBuilder{
        fatalError("NotImplementedError: addItem function not implemented")
    }

    func build() -> ShoppingBasket {
        fatalError("NotImplementedError: build function not implemented")
    }
}

class ItemBuilder {
    
    init(_ name: String) {
        fatalError("NotImplementedError: ItemBuilder initializer not implemented")
    }

    func withPrice(_ price: Double) -> ItemBuilder {
        fatalError("NotImplementedError: withPrice function not implemented")
    }

    func withQuantity(_ quantity: Int) -> ItemBuilder {
        fatalError("NotImplementedError: withQuantity function not implemented")
    }

    func build() -> BasketItem {
        fatalError("NotImplementedError: build function not implemented")
    }
}
